File: src/WorkBC.Data/Model/JobBoard/Job.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WorkBC.Data.Model.JobBoard
{
    public class Job
    {
        [DatabaseGenerated(DatabaseGeneratedOption.None)]
        [Key]
        public long JobId { get; set; }

        [ForeignKey("JobId")]
        public virtual JobId Id { get; set; }

        [StringLength(300)]
        public string Title { get; set; }

        [ForeignKey("Location")]
        public int LocationId { get; set; }

        public virtual Location Location { get; set; }

        [StringLength(120)]
        public string City { get; set; }

        [StringLength(100)]
        public string EmployerName { get; set; }

        /// <summary>
        ///     2016 Noc Code
        /// </summary>
        public short? NocCodeId { get; set; }

        [ForeignKey("NocCodeId")]
        public virtual NocCode NocCode { get; set; }

        /// <summary>
        ///     2021 Noc Code
        /// </summary>
        public int? NocCodeId2021 { get; set; }

        [ForeignKey("NocCodeId2021")]
        public virtual NocCode2021 NocCode2021 { get; set; }

        /// <summary>
        ///     Industry code
        /// </summary>
        public short? IndustryId { get; set; }

        [ForeignKey("IndustryId")]
        public virtual Industry Industry { get; set; }

        /// <summary>
        ///     Salary amount
        /// </summary>
        [Column(TypeName = "decimal(18,2)")]
        public decimal? Salary { get; set; }

        /// <summary>
        ///     Salary summary string to appear in saved job results
        /// </summary>
        [StringLength(60)]
        public string SalarySummary { get; set; }

        public short PositionsAvailable { get; set; }

        public DateTime DatePosted { get; set; }

        public DateTime ActualDatePosted { get; set; }

        public DateTime ExpireDate { get; set; }

        public DateTime LastUpdated { get; set; }

        /// <summary>
        ///     The first time our importers found a job with this job id
        /// </summary>
        public DateTime DateFirstImported { get; set; }

        /// <summary>
        ///     The date the job was last updated by the importer
        /// </summary>
        public DateTime DateLastImported { get; set; }

        public bool IsActive { get; set; }

        public bool FullTime { get; set; }

        public bool PartTime { get; set; }

        public bool LeadingToFullTime { get; set; }

        public bool Permanent { get; set; }

        public bool Temporary { get; set; }

        public bool Casual { get; set; }

        public bool Seasonal { get; set; }

        public virtual JobSource JobSource { get; set; }

        [ForeignKey("JobSource")]
        public byte JobSourceId { get; set; }

        [StringLength(100)]
        public string OriginalSource { get; set; }

        [StringLength(800)]
        public string ExternalSourceUrl { get; set; }

        [NotMapped]
        public List<string> HoursOfWork
        {
            get
            {
                var list = new List<string>();
                if (FullTime)
                {
                    list.Add("Full-time");
                }


                if (PartTime)
                {
                    list.Add("Part-time");
                }


                if (LeadingToFullTime)
                {
                    list.Add("Part-time leading to full-time");
                }

                return list;
            }
        }

        [NotMapped]
        public List<string> PeriodOfEmployment
        {
            get
            {
                var list = new List<string>();

                if (Permanent)
                {
                    list.Add("Permanent");
                }

                if (Temporary)
                {
                    list.Add("Temporary");
                }

                if (Casual)
                {
                    list.Add("Casual");
                }

                if (Seasonal)
                {
                    list.Add("Seasonal");
                }

                return list;
            }
        }

        [NotMapped]
        public ExternalJobSource ExternalSource =>
            new ExternalJobSource
            {
                Source = new List<ExternalSource>
                {
                    new ExternalSource {Source = OriginalSource, Url = ExternalSourceUrl}
                }
            };

    }

    [NotMapped]
    public class ExternalJobSource
    {
        public List<ExternalSource> Source { get; set; }
    }

    [NotMapped]
    public class ExternalSource
    {
        public string Url { get; set; }
        public string Source { get; set; }
    }
}

File: app/Models/Job.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

final class Job extends Model
{
    protected $table = 'Jobs';

    protected $primaryKey = 'JobId';

    protected $keyType = 'string';

    public $incrementing = false;

    public $timestamps = false;

    protected $casts = [
        'FullTime' => 'boolean',
        'PartTime' => 'boolean',
        'LeadingToFullTime' => 'boolean',
        'Permanent' => 'boolean',
        'Temporary' => 'boolean',
        'Casual' => 'boolean',
        'Seasonal' => 'boolean',
        'IsActive' => 'boolean',
        'DatePosted' => 'datetime',
        'DateCreated' => 'datetime',
        'DateModified' => 'datetime',
        'ExpireDate' => 'datetime',
        'LastUpdated' => 'datetime',
        'Salary' => 'decimal:2',
    ];

    public function jobSource(): BelongsTo
    {
        return $this->belongsTo(JobSource::class, 'JobSourceId', 'Id');
    }

    public function location(): BelongsTo
    {
        return $this->belongsTo(Location::class, 'LocationId', 'LocationId');
    }

    // NOTE: Jobs has no RegionId column — a job's region is reached through its
    // Location ($job->location?->region; Locations.RegionId → Regions.Id), or read
    // from the OpenSearch Region field for search.

    public function industry(): BelongsTo
    {
        return $this->belongsTo(Industry::class, 'IndustryId', 'Id');
    }

    public function nocCode(): BelongsTo
    {
        return $this->belongsTo(NocCode::class, 'NocCodeId', 'Id');
    }

    public function nocCode2021(): BelongsTo
    {
        return $this->belongsTo(NocCode2021::class, 'NocCodeId2021', 'Id');
    }

    public function savedJobs(): HasMany
    {
        return $this->hasMany(SavedJob::class, 'JobId', 'JobId');
    }
}
